profiling/src/profiling/mod.rs

```diff
@@ -665,6 +665,54 @@ impl Profiler {
         }
     }
 
+    #[cfg(feature = "timeline")]
+    pub unsafe fn collect_compile_file(
+        &self,
+        duration: i64,
+        filename: String,
+        include_type: &str,
+        locals: &RequestLocals,
+    ) {
+        let mut labels = Profiler::message_labels();
+
+        lazy_static! {
+            static ref TIMELINE_COMPILE_FILE_LABELS: Vec<Label> = vec![Label {
+                key: "event",
+                value: LabelValue::Str("compilation".into()),
+            },];
+        }
+
+        labels.extend_from_slice(&TIMELINE_COMPILE_FILE_LABELS);
+        labels.push(Label {
+            key: "filename",
+            value: LabelValue::Str(Cow::from(filename)),
+        });
+        let n_labels = labels.len();
+
+        match self.send_sample(Profiler::prepare_sample_message(
+            vec![ZendFrame {
+                function: format!("[{include_type}]"),
+                file: None,
+                line: 0,
+            }],
+            SampleValues {
+                timeline: duration,
+                ..Default::default()
+            },
+            labels,
+            locals,
+        )) {
+            Ok(_) => {
+                trace!("Sent event 'compile file' with {n_labels} and labels to profiler.")
```

Comment:
There's a superfluous `and` here:

```suggestion
                trace!("Sent event 'compile file' with {n_labels} labels to profiler.")
```